java/src/main/native/src/row_conversion.cu

```diff
@@ -163,6 +167,11 @@ struct tile_info {
 
   __device__ inline size_type get_shared_row_size(size_type const *const col_offsets,
                                                   size_type const *const col_sizes) const {
+    // this calculation is invalid if there are holes in the data such as a variable-width column.
```

Comment:
This calculation is asking "How much data is there between column x and y?" so that it can calculate the shared memory requirements for a row. This works 100% for fixed-width data, but with variable-width data we don't actually fill the fixed-width portion(length/offset) so we are going to read and write some garbage for this data. This isn't destructive, but it isn't a perfect use of the bandwidth. The thought was that it would be more work to try and skip these variable width column "holes" than to just allow the overscan/overwrite.